megamek/src/megamek/client/ui/swing/tileset/TilesetManager.java

```diff
@@ -0,0 +1,556 @@
+/*
+* MegaMek -
+* Copyright (C) 2002, 2003, 2004 Ben Mazur (bmazur@sev.org)
+* Copyright (C) 2018, 2020 The MegaMek Team
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU General Public License as published by the Free Software
+* Foundation; either version 2 of the License, or (at your option) any later
+* version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+* details.
+*/
+
+package megamek.client.ui.swing.tileset;
+
+import java.util.*;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Image;
+import java.awt.MediaTracker;
+import java.awt.Polygon;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import megamek.client.ui.ITilesetManager;
+import megamek.client.ui.swing.GUIPreferences;
+import megamek.client.ui.swing.boardview.BoardView1;
+import megamek.client.ui.swing.tileset.MechTileset.MechEntry;
+import megamek.client.ui.swing.util.ImageCache;
+import megamek.client.ui.swing.util.ImageFileFactory;
+import megamek.client.ui.swing.util.PlayerColors;
+import megamek.common.*;
+import megamek.common.preference.*;
+import megamek.common.util.DirectoryItems;
+import megamek.common.util.ImageUtil;
+import megamek.common.util.MegaMekFile;
+
+/**
+ * Handles loading and manipulating images from both the mech tileset and the
+ * terrain tileset.
+ *
+ * @author Ben
+ */
+public class TilesetManager implements IPreferenceChangeListener, ITilesetManager {
```

Comment:
Did you just move this file, or did you make changes to it? If the latter, could you point to where the changes are?

Reply:
I'm really a git noob. Does it show everything as changed because I moved it?
Well in reality, I changed almost everything because there was so much redundant and (to me) unorganized code. I also moved out EntityImage to its own class and removed the default hex Tileset that was in there as a 1000-line long list of println (instead that default is now present as a simple text file in case something with the actual tileset files goes wrong... and I guess there's a few other files that can crash MM when they're not present so why worry too much about this one).

Reply:
Ok, I'll review it as "fresh".